docs/DEPLOY.md

```diff
@@ -257,92 +257,206 @@ There is an example nginx configuration provided at `config/deploy/nginx.conf` a
+
+```scheme
+(use-modules (bonfire services bonfire) ;for 'oci-bonfire-service-type'
+             (oci services meilisearch) ;for 'oci-meilisearch-service-type'
+             ...)
+
+...
+
+;; meilisearch
+(service oci-meilisearch-service-type
+         (oci-meilisearch-configuration
+          (network "host")
+          (master-key "/path/to/meilisearch/master/key/secret")))
+
+;; Bonfire
+(service oci-bonfire-service-type
+         (oci-bonfire-configuration
+          (configuration
+           (bonfire-configuration
+            ;; Networking
+            (hostname "yourdomain.net")  ;replace with your domain
+            (public-port "443")
+            (network "host")
+            ;; Postgres
+            (postgres-user "bonfire")
+            (postgres-db "bonfire")
+            ;; Mail
+            (mail-domain "yourdomain.net")
+            (mail-from "youremail@address.org")))
+          ;; Shepherd dependencies
+          (requirement
+           '(user-processes postgresql postgres-roles podman-meilisearch))
+          (extra-variables
+           `(("MAIL_BACKEND" . "mailjet")
+             ("SERVER_PORT" . "4000")
+             ("SEARCH_MEILI_INSTANCE" . "http://localhost:7700")))
+          ;; Secrets
+          (meili-master-key
+           "/path/to/meilisearch/master/key/secret")
+          (postgres-password
+           "/path/to/postgres/password/secret")
+          (mail-key
+           "/path/to/mail/key/secret")
+          (mail-private-key
+           "/path/to/mail/private/key/secret")
+          (secret-key-base
+           "/path/to/secret/key/base/secret")
+          (signing-salt
+           "/path/to/signing/salt/secret")
+          (encryption-salt
+           "/path/to/encryption/salt/secret")))
+```
+
+One option to provision secrets as files in a Guix aware way is [sops-guix](https://github.com/fishinthecalculator/sops-guix).
+
+#### 6. Reverse proxy
+
+The last piece to be able to access your instance from the Internet is a reverse proxy. We'll use NGINX but any one should work. To configure NGINX to forward traffic to Bonfire the following must be added to your configuration:
+
+```scheme
+(use-modules (gnu services web) ;for 'nginx-service-type'
+             ...)
+
+...
+
+(service nginx-service-type
+         (nginx-configuration
+          ;; Wait for Bonfire to start
+          (shepherd-requirement '(podman-bonfire))
+          (server-blocks
+           (list
+            (nginx-server-configuration
+             (server-name (list "yourdomain.net"))
+             (listen '("443 ssl" "[::]:443 ssl"))
+             ;; Replace with your domain
+             (ssl-certificate "/etc/certs/yourdomain.net/fullchain.pem")
+             ;; Replace with your domain
+             (ssl-certificate-key "/etc/certs/yourdomain.net/privkey.pem")
+             (locations
+              (list
+               (nginx-location-configuration
```

Comment:
have you had a look at `config/deploy/nginx.conf` in bonfire-app repo? it's just an example but the main thing is that it serves static assets from priv/static directly and only proxied any non-existing paths to the elixir app

Reply:
ah I guess not so simple if nginx is running on the host and bonfire is running in a container? what about uploads though, if they're stored on the host nginx could at least serve /data/uploads/ direcly (that's not in the sample nginx config but is in the Caddyfile)

Reply:
both should be possible with some fiddling with file system permissions, at least I was able to do it for tandoor. let me try https://codeberg.org/fishinthecalculator/guix-deployments/src/branch/main/modules/fishinthecalculator/virtual-nellone/services/nginx.scm#L64